lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -52,15 +36,6 @@ class RpcSession implements Disposable {
         this.#ec = ec
         this.#uri = uri
 
-        // We setup a recurring timer to keep the RPC session alive.
-        this.#keepAliveInterval = window.setInterval(() => {
```

Comment:
Hypothetically, other editors with different webview implementations may be able to keep the page running and wouldn't need to implement this in the extension. Thus I would keep this code around as it doesn't hurt to send more keep-alives then necessary. I would also note in `EditorApi` that the RPC session management only has to work if the webview tends to go to sleep.

Reply:
The original motivation for the keep-alive approach was that the infoview doesn't get a close event in vscode and the keep-alive was the easiest way to close the session reliably.  This didn't work out, so we need to do this differently and ultimately no longer need keep-alives.  As you've pointed out, other editors might even be able to handle this in a cleaner way.

> other editors with different webview implementations may be able to keep the page running

I believe that's the standard behavior in electron, so we'll encounter it in other editors as well.

> I would also note in EditorApi that the RPC session management only has to work if the webview tends to go to sleep.

There's no supported way to detect this.  Note that the infoview will still respond to messages, it's only the setInterval call which is throttled.

The reason for this throttling is ultimately to reduce CPU and power usage by reducing the number of unnecessary wake-ups.  I don't want to work against that goal if there's a straightforward alternative.  It's not the highest priority, but having a lean file open in the background shouldn't reduce the battery life of my laptop.

Reply:
> The original motivation for the keep-alive approach was that the infoview doesn't get a close event in vscode and the keep-alive was the easiest way to close the session reliably. This didn't work out, so we need to do this differently and ultimately no longer need keep-alives.

The original motivation was nothing specific to VSCode, it is that we are building a distributed system in which the client may randomly break/loop/catch fire, and we want the server to still clean up its resources in any such case without relying on the client to behave correctly. I don't understand the claim that it "didn't work". It's unfortunate that we have to move the timer to the extension in order to maintain liveness, but it will work there, no?

> I believe that's the standard behavior in electron

Yes, but there are also webkit-based webviews and others.

> There's no supported way to detect this.

What I meant was not a dynamic check but rather an observation/specification of how a particular webview behaves will determine where in the code the timer needs to live.

> The reason for this throttling is ultimately to reduce CPU and power usage by reducing the number of unnecessary wake-ups. I don't want to work against that goal if there's a straightforward alternative. It's not the highest priority, but having a lean file open in the background shouldn't reduce the battery life of my laptop.

I absolutely agree, but note that this came from web browsers in which a malicious website can set up high-frequency timers/bitcoin miners/whatever. And with many tabs open, it makes sense to put old ones to sleep. We are not doing any of that. In any sane implementation, a 10s-interval timer will use close to no CPU or power, so I don't think this is a concern.

Reply:
Update: we decided to remove the keepalive code from infoview for now. We can readd it if it is ever needed for another editor.